roles/cleanup-cinder-csi-volumes/files/cleanup_volumes.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+"""
+This script removes volumes created by cinder csi that are
+* older than 3 hours
+* not attached to any node
+"""
+import openstack
+import logging
+import sys
+from datetime import datetime, timedelta
+
+def main():
+
+    logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+    log = logging.getLogger(__name__)
+    log.info("cleaning up orphaned volumes")
+
+    con = openstack.connect()
+    volumes = con.block_storage.volumes(details=True, status="available")
+
+    for v in filter(filter_volumes, volumes):
+        log.info("deleting volume %s", v.id)
+        try:
+            con.block_storage.delete_volume(v, ignore_missing=True)
+        except:
+            log.exception("unable to delete volume %s... skipping", v.id)
+            pass
+
+def filter_volumes(vol):
```

Comment:
@wangxiyuan @bzhaoopenstack please take a look at the function here to make sure it doesn't remove volumes that shouldn't be removed.

Reply:
Thanks, I think this function is OK  to  go. As current volumes which is used in current env are not created with the name prefix "pvc-", and if the volume is using by other VM, those ones can not be deleted.